tox_packet/src/dht/cookie.rs

```diff
@@ -160,10 +162,11 @@ impl EncryptedCookie {
         }
     }
     /// Calculate SHA512 hash of encrypted cookie together with nonce
-    pub fn hash(&self) -> sha512::Digest {
+    pub fn hash(&self) -> [u8; <Sha512 as Digest>::OutputSize::USIZE] {
         let mut buf = [0; 112];
         let (_, size) = self.to_bytes((&mut buf, 0)).unwrap();
-        sha512::hash(&buf[..size])
+        // TODO: use `Into` directly when GenericArray supports it
```

Comment:
no need to, since generic array will be removed because of const generics

Reply:
It's unlikely - const generics are less powerful. It could be [replaced by another library](https://github.com/fizyk20/generic-array/issues/115#issuecomment-808272099) but until then worth to have this todo, otherwise after migration wi might miss this useless transformation.